src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -568,22 +577,42 @@ public synchronized void applyDML(Sequenced<DMLEvent> sequencedEvent) throws Exc
 
   @VisibleForTesting
   synchronized void flush() throws InterruptedException, IOException, DeltaFailureException {
-    Collection<TableBlob> tableBlobs;
+    Map<MultiGCSWriter.BlobType, Collection<TableBlob>> tableBlobsByBlobType;
     // if this throws an IOException, we want to propagate it, since we need the app to reset state to the last
     // commit and replay events. This is because previous events are written directly to an outputstream to GCS
     // and then dropped, so we cannot simply retry the flush here.
     try {
-      tableBlobs = gcsWriter.flush();
+      tableBlobsByBlobType = gcsWriter.flush();
     } catch (IOException e) {
       flushException = e;
       throw e;
     }
 
+    processBlobsInParallel(tableBlobsByBlobType.get(MultiGCSWriter.BlobType.SNAPSHOT));
+    processBlobsInParallel(tableBlobsByBlobType.get(MultiGCSWriter.BlobType.STREAMING));
+
+    latestMergedSequence.clear();
+    latestMergedSequence.putAll(latestSeenSequence);
+    commitOffset();
+  }
+
+  private void processBlobsInParallel(Collection<TableBlob> tableBlobs)
+    throws InterruptedException, DeltaFailureException {
     List<Future<?>> mergeFutures = new ArrayList<>(tableBlobs.size());
     for (TableBlob blob : tableBlobs) {
       // submit a callable instead of a runnable so that it can throw checked exceptions
       mergeFutures.add(executorService.submit((Callable<Void>) () -> {
-        mergeTableChanges(blob);
+        if (blob.isSnapshotOnly()) {
+          context.putState(String.format(DIRECT_LOADING_IN_PROGRESS_PREFIX + "%s-%s", blob.getDataset(),
```

Comment:
does this get cleared or set to false anywhere?

Reply:
yes it does get set to false in the else block below on line number 611